manifests/interface.pp

```diff
@@ -16,7 +16,6 @@
 # @param peers is an array of struct (Wireguard::Peers) for multiple peers
 # @param routes different routes for the systemd-networkd configuration
 # @param private_key Define private key which should be used for this interface, if not provided a private key will be generated
-# @param preshared_key Define preshared key which should be used for this interface
```

Comment:
This should be possible also, we have public_key and endpoint also as parameter. with these parameters we build a peer object and append it to the peers array (this was implemented to prevent breaking changes) see:
https://github.com/voxpupuli/puppet-wireguard/blob/5859306deddd56a28edc50a15ba9918aab863efe/manifests/interface.pp#L163-L167